### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by ragangithub**
- Created a method called current_user in  ApplicationController to make current user data available to all controllers..
- Created forms to perform the following functions:
    - Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
    - Creates a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Allowed Users to add likes to Posts.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -45,5 +45,9 @@
     <button class="back-all-users">
       <%= link_to "Back to all users", root_path %>
     </button>
```
Comment (brainconnect93) [inline app/views/posts/index.html.erb:51]: - [ ] Great effort working on the project milestone but currently I could not access the form for posts and comments which makes it difficult to create either post or comment. I would kindly suggest you work on the error message as i have recorded a video to demonstrate how the app is currently behaving.

Video demonstration 👇 

[screen-capture (3).webm](https://github.com/ragangithub/Blog_app/assets/78634841/b08925ad-8aa0-435d-9db8-4aa374a51c65)

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @ragangithub, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight :1st_place_medal: 

-  Descriptive Pull Request title and summary. ✔️  
- No Linter errors :heavy_check_mark: 
- Good use of Git flow ✔️ 
- Blog app - add forms ✔️ 

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ragangithub** [comment]
- Fixed the error caused by link for pagination preventing  adding posts and comments

**Review by franclobo — APPROVED** [review]
Hi @ragangithub, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Use preprocessed HTML file with embedded Ruby code. ✔️
- Use layouts and templates for shared content. ✔️

## Optional suggestions

- **[OPTIONAL]** Please add styles to be similar to the wireframes for a better UI. 
- **[OPTIONAL]** We could use bootstrap forms to give styles to the forms for a better UI.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**